src/analysis/modifications/generic.jl

```diff
@@ -65,3 +65,55 @@ change_objective(
         v = opt_model[:x]
         @objective(opt_model, sense, sum(opt_weights[i] * v[i] for i in objective_indices))
     end
+
+"""
+    add_crowding(weights::Vector{Float64}, idxs::Vector{Float64})
+
+Adds a molecular crowding constraint to the optimization problem: `∑ wᵢ × vᵢ ≤ 1` where `wᵢ`
+is an element in `weights` and `vᵢ` is a flux with index in the model's reactions given by
+`idxs`.
+
+See `Beg, Qasim K., et al. "Intracellular crowding defines the mode and sequence of
+substrate uptake by Escherichia coli and constrains its metabolic activity." Proceedings of
+the National Academy of Sciences 104.31 (2007): 12663-12668.` for more details.
+"""
+add_crowding(weights::Vector{Float64}, idxs::Vector{Int64}) =
```

Comment:
what about passing in a Dict{String,Float64} ?

Reply:
ah there are more versions below. It might be nicer to allow using the structured data instead of matlabish "vectors that must correspond".